src/api/c/blas.cpp

```diff
@@ -145,23 +145,32 @@ af_err af_matmul(af_array *out, const af_array lhs, const af_array rhs,
         }
 
         TYPE_ASSERT(lhs_type == rhs_type);
-        af_array output = 0;
+        // af_array output = 0;
+        int aRowDim = (optLhs == AF_MAT_NONE) ? 0 : 1;
+        int bColDim = (optRhs == AF_MAT_NONE) ? 1 : 0;
+        int M       = lDims[aRowDim];
+        int N       = rDims[bColDim];
+        dim_t d2    = std::max(lDims[2], rDims[2]);
+        dim_t d3    = std::max(lDims[3], rDims[3]);
+        const dim4 outDims(M, N, d2, d3);
```

Comment:
I am thinking if this output size calculation should be abstracted out into respective backends. If not, at least put in `src/backend/common` instead of repeating this code every where.

We should use something like the following
```cpp
dim4 blasOutputDims(dim4 lhsDims, dim4 rhsDims, matProp optLhs, matProp optRhs);
```

Reply:
Thanks, this was actually one thing that I was iffy about too, but I decided that time to move on temporarily in order to address the failing tests. In any case, coming back to this issue now - my first attempt to resolve this is e9fada9 - please let me know your thoughts.